addons/account/views/account_tax_views.xml

```diff
@@ -40,11 +40,15 @@
                     <field name="factor_percent" attrs="{'invisible': [('repartition_type', '=', 'base')]}"/>
                     <field name="repartition_type"/>
                     <field name="account_id" attrs="{'invisible': [('repartition_type', '=', 'base')]}" options="{'no_create': True}"/>
-                    <field name="tag_ids" widget="many2many_tags" options="{'no_create': True}" domain="[('applicability', '=', 'taxes'), '|', ('country_id', '=', parent.country_id), ('country_id', '=', False)]"/>
+                    <field name="tag_ids"
+                           widget="many2many_tags"
+                           options="{'no_create': True}"
+                           domain="tag_ids_domain"/>
```

Comment:
I know, it's crazy! I haven't seen anything like that in Odoo itself.

I figured out this solution by discussing with WAN. He suggested referencing another field without knowing if that would work but that make sense because we do that kind of reference for other stuff like the `attrs`.

Using a `fields.Char` doesn't work  as it doesn't get interpreted as a domain but as a String which miserably fails.
I think the idea of using a `fields.Binary` is inspired by one of the early iteration Habib has done for it's analytic widget. He stored a dictionary in a `fields.Binary`

Googling "odoo dynamic domain", you can also find [this topic](https://www.odoo.com/fr_FR/forum/aide-1/how-to-create-dynamic-domain-on-many2one-fields-with-onchange-function-30742) in which Niyasraphi mention [this OCA module](https://odoo-community.org/shop/web-domain-field-4368#attr=22677) that has made that possible using `fields.Char` storing a `json.dump(domain)` since version 10.0 :exploding_head:

Maybe using it with a Binary field has always been possible :exploding_head: